tictactoe/adapter-implementation/src/main/kotlin/com/anaplan/engineering/azuki/tictactoe/adapter/implementation/query/SampleQueryFactory.kt

```diff
@@ -0,0 +1,64 @@
+import com.anaplan.engineering.azuki.core.system.ensureRunnable
+import com.anaplan.engineering.azuki.core.system.unsupportedBehavior
+import com.anaplan.engineering.azuki.tictactoe.adapter.api.PlayerCheckFactory
+import com.anaplan.engineering.azuki.tictactoe.adapter.api.Position
+import com.anaplan.engineering.azuki.tictactoe.adapter.api.TicTacToeCheckFactory
+import com.anaplan.engineering.azuki.tictactoe.adapter.api.TicTacToeQueryFactory
+import com.anaplan.engineering.azuki.tictactoe.adapter.implementation.ExecutionEnvironment
+import com.anaplan.engineering.azuki.tictactoe.adapter.implementation.canMove
+import com.anaplan.engineering.azuki.tictactoe.adapter.implementation.tokenAt
+
+class SampleQueryFactory : TicTacToeQueryFactory {
+
+    override fun <T, C : Collection<T>> createForAllQuery(
+        derivedFrom: (TicTacToeQueryFactory) -> Query<C>, deriveQuery: (T, TicTacToeQueryFactory) -> DerivedQuery<*>
+    ) = ForAllRunnableDerivedQuery(derivedFrom(this).ensureRunnable()) { t ->
+        @Suppress("UNCHECKED_CAST") (deriveQuery(t,
+            this) as RunnableDerivedQuery<ExecutionEnvironment, TicTacToeCheckFactory, *>)
+    }
+
+    override fun <T> liftQueriesToDerivedQuery(queries: List<Query<*>>) =
+        ListRunnableDerivedQuery.fromQueries<ExecutionEnvironment, TicTacToeCheckFactory, T>(queries)
+
+    override fun getPlayOrder(gameName: String) = query(value = { env ->
+        env.withGame(gameName) { playOrder.map { it.token.symbol } }
+    }, checks = { playOrder -> { listOf(game.hasPlayOrder(gameName, playOrder)) } })
+
+    override fun getPositions(gameName: String) = query(value = { env ->
+        env.withGame(gameName) { (1..height).flatMap { row -> (1..width).map { col -> Position(row, col) } } }
+    })
+
+    override fun getToken(gameName: String, position: Position) =
+        query(value = { env -> env.withGame(gameName) { tokenAt(position)?.symbol } }, checks = { token ->
+            token?.let { { listOf(game.hasToken(gameName, it, position)) } } ?: {
+                listOf(game.hasSpace(gameName, position))
+            }
+        })
+
+    override fun canPlayerPlaceToken(gameName: String, playerName: String, position: Position) = query(value = { env ->
+        env.withGame(gameName) { canMove(playerName, position) }
+    }, checks = { canPlace ->
+        val check = if (canPlace) PlayerCheckFactory::canPlaceToken else PlayerCheckFactory::cannotPlaceToken
+        { listOf(check(player, gameName, playerName, position)) }
+    })
+}
+
+private fun <T> query(
+    value: (ExecutionEnvironment) -> T,
+    checks: (T) -> TicTacToeCheckFactory.() -> List<Check> = { { listOf(UnsupportedCheck) } },
+): Query<T> = object : RunnableQuery<ExecutionEnvironment, TicTacToeCheckFactory, T> {
+
+    override val behavior: Behavior get() = unsupportedBehavior
+    override fun run(environment: ExecutionEnvironment) = value(environment).let { SampleAnswer(this, it, checks(it)) }
+}
```

Comment:
This could also be lifted out, similar to the way I considered making `answer(..)` a function.

I've curried `checks` here because I like the ability to partially apply it into `SampleAnswer`, but I realise it makes everything else a bit ugly, so I'm happy to back the currying back out if needed.